.golangci.yml

```diff
@@ -0,0 +1,26 @@
+run:
+  skip-dirs:
+  - .*/mocks
+
+issues:
+  # https://github.com/golangci/golangci-lint/issues/2439
+  exclude-use-default: false
+
+linters:
+  enable:
+  - errcheck
+  - gosimple
+  - govet
+  - ineffassign
+  - staticcheck
+  - typecheck
+  - unused
+  - revive
+
+linters-settings:
+  revive:
+    severity: error
+    rules:
+    - name: exported
+      arguments:
+      - checkPrivateReceivers
```

Comment:
This setting is exactly the same as in your other PR:
https://github.com/bitrise-io/go-utils/pull/168

Do we need to standardise it like what we are doing with the yamlint settings? The `steps-check` step contains the default settings and writes it to a file before running that given check.

Reply:
Hard to say, my argument for duplicating it across the few library repos is that we only have 3 such repos at the moment and I don't expect to create new ones in the future. 

Are you thinking about creating a new step, such as `libs-check`, or extending the `steps-check` code to apply different golangci-lint rules for steps and libs?

Reply:
I was thinking of somehow adding this config yaml to the `steps-check` step and somehow configuring when to apply it. But now that I think about it we can do it later as well and it will be easy to do it even if we merge this PR as it is.

One of the CI seems to have failed while the other succeeded so I think probably just restarting the build probably will solve it.

Reply:
Agree, let's do the abstraction later. I restarted the CI build, it was indeed just flakiness. The only remaining thing is a ✅ from you 🥺 😁